src/os/ObjectStore.h

```diff
@@ -1665,123 +1393,73 @@ class ObjectStore {
       uint64_t expected_write_size,
       uint32_t flags
     ) {
-      if (use_tbl) {
-        __u32 op = OP_SETALLOCHINT;
-        ::encode(op, tbl);
-        ::encode(cid, tbl);
-        ::encode(oid, tbl);
-        ::encode(expected_object_size, tbl);
-        ::encode(expected_write_size, tbl);
-      } else {
-        Op* _op = _get_next_op();
-        _op->op = OP_SETALLOCHINT;
-        _op->cid = _get_coll_id(cid);
-        _op->oid = _get_object_id(oid);
-        _op->expected_object_size = expected_object_size;
-        _op->expected_write_size = expected_write_size;
-	_op->alloc_hint_flags = flags;
-      }
+      Op* _op = _get_next_op();
+      _op->op = OP_SETALLOCHINT;
+      _op->cid = _get_coll_id(cid);
+      _op->oid = _get_object_id(oid);
+      _op->expected_object_size = expected_object_size;
+      _op->expected_write_size = expected_write_size;
+      _op->alloc_hint_flags = flags;
       data.ops++;
     }
 
     void encode(bufferlist& bl) const {
-      if (use_tbl) {
-        uint64_t ops = data.ops;
-        uint64_t pad_unused_bytes = 0;
-        uint32_t largest_data_len = data.largest_data_len;
-        uint32_t largest_data_off = data.largest_data_off;
-        uint32_t largest_data_off_in_tbl = data.largest_data_off_in_tbl;
-        bool tolerate_collection_add_enoent = false;
-	uint32_t fadvise_flags = data.fadvise_flags;
-        ENCODE_START(8, 5, bl);
-        ::encode(ops, bl);
-        ::encode(pad_unused_bytes, bl);
-        ::encode(largest_data_len, bl);
-        ::encode(largest_data_off, bl);
-        ::encode(largest_data_off_in_tbl, bl);
-        ::encode(tbl, bl);
-        ::encode(tolerate_collection_add_enoent, bl);
-	::encode(fadvise_flags, bl);
-        ENCODE_FINISH(bl);
-      } else {
-        //layout: data_bl + op_bl + coll_index + object_index + data
-        ENCODE_START(9, 9, bl);
-        ::encode(data_bl, bl);
-        ::encode(op_bl, bl);
-        ::encode(coll_index, bl);
-        ::encode(object_index, bl);
-        data.encode(bl);
-        ENCODE_FINISH(bl);
-      }
+      //layout: data_bl + op_bl + coll_index + object_index + data
+      ENCODE_START(9, 9, bl);
+      ::encode(data_bl, bl);
+      ::encode(op_bl, bl);
+      ::encode(coll_index, bl);
+      ::encode(object_index, bl);
+      data.encode(bl);
+      ENCODE_FINISH(bl);
     }
+
     void decode(bufferlist::iterator &bl) {
-      DECODE_START_LEGACY_COMPAT_LEN(9, 5, 5, bl);
-      DECODE_OLDEST(2);
-
-      bool decoded = false;
-      if (struct_v < 8) {
-	decode8_5(bl, struct_v);
-	use_tbl = true;
-	decoded = true;
-      }	else if (struct_v == 8) {
+      DECODE_START(9, bl);
+      DECODE_OLDEST(8);
+
+      /* Actual version should be 9, but some version 9
+       * transactions ended up with version 8 */
+      if (struct_v == 8) {
 	bufferlist::iterator bl2 = bl;
 	try {
 	  decode8_5(bl, struct_v);
-	  use_tbl = true;
-	  decoded = true;
+	  assert(0 == "Now not support this format");
```

Comment:
v8 -> v9 bug was fixed in 2598fc50749f7a1e8450c07b27cec5fa54b3e41f, which precedes jewel, so v8 support is safe to remove entirely.